self-api/src/main/java/com/selfxdsd/api/Projects.java

```diff
@@ -51,4 +53,22 @@
      * @return Projects.
      */
     Projects ownedBy(final User user);
+
+    /**
+     * Get the Project with the corresponding projectId.
+     * @param projectId Id of the Project.
+     * @return Project, or null if no Project with this id is found.
+     */
+    default Project getProjectById(final int projectId) {
```

Comment:
@BogdanAdrian11 This method cannot be default, because the base implementation of ``Projects`` will represent *all* projects in the system, and we will not be able to iterate over all of them (loading them all into memory will probably be not possible). 

But it's ok to leave it like this for now. Just specify in the ``TODO`` that this method should be implemented by all implementors of Projects.